recipes/ruby/all/test_package/conanfile.py

```diff
@@ -1,21 +1,55 @@
-from conan import ConanFile
-from conan.tools.cmake import CMake
-from conans import tools
 import os
+import re
+from io import StringIO
+
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, cmake_layout
+
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "CMakeDeps", "CMakeToolchain"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeToolchain", "CMakeDeps", "VirtualRunEnv"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str, run=True, libs=True)
+
+    def layout(self):
+        cmake_layout(self)
 
     def build(self):
         cmake = CMake(self)
-        cmake.configure()
+        # when --static-linked-ext is used, ruby defines EXTSTATIC as 1
+        # But when ruby itself is static there's nothing, so:
+        # We define RUBY_STATIC_RUBY when ruby itself is static
+        # We define RUBY_STATIC_LINKED_EXT when the ruby extensions are static (same as EXTSTATIC but clearer)
+        defs = {}
+        tested_options = self.dependencies[self.tested_reference_str].options
+        if not tested_options.shared:
+            defs["RUBY_STATIC_RUBY"] = 1
+            if tested_options.with_static_linked_ext:
+                defs["RUBY_STATIC_LINKED_EXT"] = 1
```

Comment:
Shouldn't these be added to `self.cpp_info.defines` in `package_info()` instead?

Reply:
I honestly have no idea... you tell me! What I'm trying to do is to pass it to the CMakeLists.txt so they have defined in the executable: https://github.com/jmarrec/conan-center-index/blob/e5c6c958b0b90067ec7d93a468405095f4eff8be/recipes/ruby/all/test_package/CMakeLists.txt#L14-L20

And then: https://github.com/jmarrec/conan-center-index/blob/e5c6c958b0b90067ec7d93a468405095f4eff8be/recipes/ruby/all/test_package/test_package.cpp#L17-L27